app/app.py

```diff
@@ -340,6 +398,94 @@ def init_db(
     )
 
 
+async def init_lifespan(
+    app: FastAPI,
+    settings: Settings,
+    hyperion_error_logger: logging.Logger,
+) -> tuple[Scheduler, WebsocketConnectionManager]:
+    # We need to run the factories and the google api credentials only once across all the workers
+    # We use the parent process to get the workers
+    parent_pid = os.getppid()  # PID du parent (FastAPI master process)
+    parent_process = psutil.Process(parent_pid)
+    workers = [
+        p for p in parent_process.children() if p.status() != psutil.STATUS_ZOMBIE
+    ]
+
+    # There is more than one worker
+    # We use the redis client to check if the factories have already been run and the google api credentials have been set
+    if len(workers) > 1:
```

Comment:
Do we really gain from checking this? I think we could directly check if a Redis is available